scipy/integrate/tests/test_quadrature.py

```diff
@@ -374,88 +373,99 @@ def test_array_like(self):
+            qmc_quad(lambda x: 1, a, b, n_estimates=8.5)
 
         message = "`qrng` must be an instance of scipy.stats.qmc.QMCEngine."
         with pytest.raises(TypeError, match=message):
-            qmc_quad(lambda x: 1, [0, 0], [1, 1], qrng="a duck")
+            qmc_quad(lambda x: 1, a, b, qrng="a duck")
 
         message = "`qrng` must be initialized with dimensionality equal to "
         with pytest.raises(ValueError, match=message):
-            qmc_quad(lambda x: 1, [0, 0], [1, 1], qrng=stats.qmc.Sobol(1))
+            qmc_quad(lambda x: 1, a, b, qrng=stats.qmc.Sobol(1))
 
         message = r"`log` must be boolean \(`True` or `False`\)."
         with pytest.raises(TypeError, match=message):
-            qmc_quad(lambda x: 1, [0, 0], [1, 1], log=10)
+            qmc_quad(lambda x: 1, a, b, log=10)
 
-    def basic_test(self, n_points=2**8, n_estimates=8, signs=None):
+    def basic_test(self, n_points=2**8, n_estimates=8, signs=None, xp=None):
+        dtype = xp_default_dtype(xp)
         if signs is None:
             signs = np.ones(2)
         ndim = 2
         mean = np.zeros(ndim)
         cov = np.eye(ndim)
 
         def func(x):
-            return stats.multivariate_normal.pdf(x.T, mean, cov)
+            # standard multivariate normal PDF in two dimensions
+            return xp.exp(-0.5 * xp.sum(x*x, axis=0)) / (2 * xp.pi)
 
         rng = np.random.default_rng(2879434385674690281)
         qrng = stats.qmc.Sobol(ndim, seed=rng)
         a = np.zeros(ndim)
         b = np.ones(ndim) * signs
-        res = qmc_quad(func, a, b, n_points=n_points,
-                       n_estimates=n_estimates, qrng=qrng)
+        res = qmc_quad(func, xp.asarray(a, dtype=dtype), xp.asarray(b, dtype=dtype),
+                       n_points=n_points, n_estimates=n_estimates, qrng=qrng)
         ref = stats.multivariate_normal.cdf(b, mean, cov, lower_limit=a)
         atol = special.stdtrit(n_estimates-1, 0.995) * res.standard_error  # 99% CI
-        assert_allclose(res.integral, ref, atol=atol)
+        xp_assert_close(res.integral, xp.asarray(ref, dtype=dtype), atol=atol)
         assert np.prod(signs)*res.integral > 0
 
         rng = np.random.default_rng(2879434385674690281)
         qrng = stats.qmc.Sobol(ndim, seed=rng)
-        logres = qmc_quad(lambda *args: np.log(func(*args)), a, b,
+        logres = qmc_quad(lambda *args: xp.log(func(*args)),
+                          xp.asarray(a, dtype=dtype), xp.asarray(b, dtype=dtype),
                           n_points=n_points, n_estimates=n_estimates,
                           log=True, qrng=qrng)
-        assert_allclose(np.exp(logres.integral), res.integral, rtol=1e-14)
-        assert np.imag(logres.integral) == (np.pi if np.prod(signs) < 0 else 0)
-        assert_allclose(np.exp(logres.standard_error),
-                        res.standard_error, rtol=1e-14, atol=1e-16)
+        rtol = 1e-14 if res.integral.dtype == xp.float64 else 1e-6
```

Comment:
Looks like a small tolerance bump is needed
```suggestion
        rtol = 1e-14 if res.integral.dtype == xp.float64 else 1.1e-6
```